presto-jdbc/src/test/java/io/prestosql/jdbc/BaseTestJdbcResultSet.java

```diff
@@ -42,365 +40,366 @@
-
-            assertTrue(rs.next());
-            assertEquals(rs.getLong(1), 123L);
-            assertEquals(rs.getLong(2), 456L);
-            assertEquals(rs.getLong("x"), 123L);
+        try (StatementWrapper statementWrapper = new StatementWrapper()) {
+            try (ResultSet rs = statementWrapper.getStatement().executeQuery("SELECT 123 x, 456 x")) {
+                ResultSetMetaData metadata = rs.getMetaData();
+                assertEquals(metadata.getColumnCount(), 2);
+                assertEquals(metadata.getColumnName(1), "x");
+                assertEquals(metadata.getColumnName(2), "x");
+
+                assertTrue(rs.next());
+                assertEquals(rs.getLong(1), 123L);
+                assertEquals(rs.getLong(2), 456L);
+                assertEquals(rs.getLong("x"), 123L);
+            }
         }
     }
 
     @Test
     public void testPrimitiveTypes()
-            throws Exception
+            throws SQLException
     {
-        checkRepresentation("123", Types.INTEGER, 123);
-        checkRepresentation("12300000000", Types.BIGINT, 12300000000L);
-        checkRepresentation("REAL '123.45'", Types.REAL, 123.45f);
-        checkRepresentation("1e-1", Types.DOUBLE, 0.1);
-        checkRepresentation("1.0E0 / 0.0E0", Types.DOUBLE, Double.POSITIVE_INFINITY);
-        checkRepresentation("0.0E0 / 0.0E0", Types.DOUBLE, Double.NaN);
-        checkRepresentation("true", Types.BOOLEAN, true);
-        checkRepresentation("'hello'", Types.VARCHAR, "hello");
-        checkRepresentation("cast('foo' as char(5))", Types.CHAR, "foo  ");
+        try (StatementWrapper statementWrapper = new StatementWrapper()) {
+            checkRepresentation(statementWrapper.getStatement(), "123", Types.INTEGER, 123);
+            checkRepresentation(statementWrapper.getStatement(), "12300000000", Types.BIGINT, 12300000000L);
+            checkRepresentation(statementWrapper.getStatement(), "REAL '123.45'", Types.REAL, 123.45f);
+            checkRepresentation(statementWrapper.getStatement(), "1e-1", Types.DOUBLE, 0.1);
+            checkRepresentation(statementWrapper.getStatement(), "1.0E0 / 0.0E0", Types.DOUBLE, Double.POSITIVE_INFINITY);
+            checkRepresentation(statementWrapper.getStatement(), "0.0E0 / 0.0E0", Types.DOUBLE, Double.NaN);
+            checkRepresentation(statementWrapper.getStatement(), "true", Types.BOOLEAN, true);
+            checkRepresentation(statementWrapper.getStatement(), "'hello'", Types.VARCHAR, "hello");
+            checkRepresentation(statementWrapper.getStatement(), "cast('foo' as char(5))", Types.CHAR, "foo  ");
+        }
     }
 
     @Test
     public void testDecimal()
-            throws Exception
+            throws SQLException
     {
-        checkRepresentation("0.1", Types.DECIMAL, new BigDecimal("0.1"));
-        checkRepresentation("DECIMAL '0.1'", Types.DECIMAL, new BigDecimal("0.1"));
+        try (StatementWrapper statementWrapper = new StatementWrapper()) {
```

Comment:
introduce a method:

```suggestion
        try (StatementWrapper statementWrapper = newStatement) {
```

Also, the `StatementWrapper` doesn't capture the intent (it's holder for Statement _and_ Connection)
Maybe `ConnectedStatement` would at least hint at something more than "Statement inside™"